libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -586,6 +746,56 @@ ArraySchema ArrowAdapter::tiledb_schema_from_arrow_schema(
     LOG_DEBUG(fmt::format("[ArrowAdapter] set_domain"));
     schema.set_domain(domain);
 
+    LOG_DEBUG(fmt::format(
+        "[ArrowAdapter] index_column_info length {}",
+        index_column_array->length));
+
+    // Note: this must be done after we've got the core domain, since the
+    // NDRectangle constructor requires access to the core domain.
+    if (use_current_domain) {
+        CurrentDomain current_domain(*ctx);
+        NDRectangle ndrect(*ctx, domain);
+
+        for (int64_t sch_idx = 0; sch_idx < arrow_schema->n_children;
+             ++sch_idx) {
+            auto child = arrow_schema->children[sch_idx];
+            auto type = ArrowAdapter::to_tiledb_format(child->format);
+
+            for (int64_t i = 0; i < index_column_schema->n_children; ++i) {
+                auto col_name = index_column_schema->children[i]->name;
+                if (strcmp(child->name, col_name) != 0) {
+                    continue;
+                }
+
+                if (ArrowAdapter::_isvar(child->format)) {
+                    // In the core API:
+                    //
+                    // * domain for strings must be set as (nullptr, nullptr)
+                    // * current_domain for strings cannot be set as (nullptr,
+                    //   nullptr)
+                    //
+                    // Fortunately, these are ASCII dims and we can range
+                    // these accordingly.
+                    ndrect.set_range(col_name, "\x01", "\x7f");
```

Comment:
Can you add a note too that this range sets all printable ASCII characters (I had to look it up)?

Reply:
Oh nice -- sure! :)

Reply:
Pushed